pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -541,12 +547,20 @@ func ResolvePipelineTask(
 	getTaskRun resources.GetTaskRun,
 	getRun GetRun,
 	pipelineTask v1beta1.PipelineTask,
+	pst PipelineRunState,
 ) (*ResolvedPipelineTask, error) {
 	rpt := ResolvedPipelineTask{
 		PipelineTask: &pipelineTask,
 	}
 	rpt.CustomTask = rpt.PipelineTask.TaskRef.IsCustomTask() || rpt.PipelineTask.TaskSpec.IsCustomTask()
 	numCombinations := 1
+
+	// We want to resolve all of the result references and ignore any errors at this point since there could be
+	// instances where result references are missing here, but will be later skipped or resolved in a subsequent
```

Comment:
Under what circumstances could result references be missing? In these circumstances, will pipelineTask.Matrix.CountCombinations() return the correct number of combinations, and will `resolvePipelineTask` create the correct number of TaskRuns?

I think we may try to resolve all pipeline tasks immediately, before running any. It seems to me like on the first reconcile loop, `ResolvePipelineTask` will generate `TaskRunNames` w/ length 1 for matrixed tasks fanned out from results, and on later reconcile loops will have the correct value.

I think having `TaskRunNames` in ResolvedPipelineTask doesn't work so well for matrixed pipeline tasks fanned out from array results, and would like to revisit the idea of label selectors. (We could also just merge https://github.com/tektoncd/pipeline/pull/6603 if that gets too complicated.) I think this PR is running into a similar problem as https://github.com/tektoncd/pipeline/pull/6603 though.